### PR TITLE
refactor(core): remove core program

### PR DESCRIPTION
**Pull request by grishasobol**
append `GenuineProgram` struct to `gtest`
append inner `program` struct for `core-processor`

### CODE REVIEW

File: core-processor/src/executor.rs [hunk 1]
```diff
@@ -141,18 +140,18 @@ where
     let execute = || {
         let env = Environment::new(
             ext,
-            program.code_bytes(),
+            program.code.code(),
```
Comment (breathx) [inline core-processor/src/executor.rs:143]: to consider renaming here since having `fn code` for `struct Code` looks weird

**Review by breathx — APPROVED** [review]

File: core-processor/src/executor.rs (hunk 1 above)
Comment (grishasobol) [reply core-processor/src/executor.rs:143]: This would change too many files, so I will make issue to make this in other PR

Comment (grishasobol) [reply core-processor/src/executor.rs:143]: Because we have the same method in `core::code::Code`

Comment (grishasobol) [reply core-processor/src/executor.rs:143]: https://github.com/gear-tech/gear/issues/3985

**Pull request merged**